DOCS-2131-4.1-configuration-mule-app-kt (#1026)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/intro-configuration.adoc

> include::_attributes.adoc[]

Reviewer: Curious about the Level-1 title, = Introduction to Mule 4: Configuring Mule Apps.
This is sort of two different things: is the following content a whole intro, or does it focus only on configuring apps?

Reviewer: In fact, I wonder if this topic is really focused on Advantages of Configuring Apps Using Mule 4?
Or something about using .yaml files rather than .properties files?

Is the point that although you can continue to use the .properties configuration method used in Mule 3, it is better to use the .yaml config offered in Mule 4, for the following reasons (and list them)?

Take a look at the ideas for alternative wordings below (and especially be sure that they do not alter technical meaning), but also look again at the flow of the narrative. We might be able to minimize this considerably by focusing explicitly on the advantages of .yaml and not get in to the Spring thing or other past-oriented considerations.

Reviewer: ...and now I reread your edit request and see that the focus is really...:

in Mule 3.x when you create a new application, Studio automatically adds a mule-app.properties file while in Mule 4.x this file is not created anymore. 
Clients can now create a yaml or properties file and both have to be configured using the Global Elements Configuration Properties.

So...should we shorten all of this to:

Mule 4 no longer automatically creates the application properties file that is created in Mule 3.
Instead, you have to manually config app properties or a YAML file 
The YAML file has advantages
But should you continue with properties or use yaml, you have to configure using Global Elements.
And here is the syntax for each option

endif::[]

> In Mule 3, Spring property placeholders are often used to configure apps dynamically for the environment in which they are deployed. Mule 4 contains a built-in mechanism for this that allows you to set default values and avoid the need to learn Spring. +

Reviewer: I understand the impulse to begin with Mule 3 and then tell how Mule 4 is better, but maybe we should consider getting directly to why Mule 4 is better? Maybe something like this:
```suggestion
Mule 4 contains a built-in mechanism for configuring apps dynamically for the environment in which they are deployed, enabling you to set default values in a YAML file rather than learning how to use Spring property placeholders. +
```
OR
```suggestion
Although Mule 3 required you to know how to use Spring property placeholders to configure apps dynamically for the environment in which they are deployed, Mule 4 contains a built-in mechanism that enables you to set default values in a YAML file. +
```

> Mule 3 supports `.properties` configuration files, while Mule 4 supports both `.yaml` and `.properties`  configuration files. The recommended approach is to use `.yaml` configuration files, because it allows the addition of type validations and autocompletion.

Reviewer: I'm not sure if the previous rewording suggestions are technically accurate, but if so, you could limit this para to just the one sentence:
```suggestion
Using YAML configuration files enables type validation and autocompletion, which were not possible using Mule 3 `.properties` configuration files.
```

> Example of a `.yaml` configuration file:

Reviewer: ```suggestion
Following is an example of a `.yaml` configuration file:
```

Reviewer: Not sure we need to use both "YAML file" and " `.yaml` " file.
Which is the most useful or appropriate or commonly understood?
Can we standardize to just one?

[source,yaml]
----
http:
  path: service
  port: "10000"
----

You can add the YAML file to your Mule app through the Configuration Properties Global Element. The XML configuration looks like this:
[source,xml,linenums]
----
<configuration-properties file="myConfiguration.yaml" />
----

> In Mule 3, when you create a new application via Anypoint Studio, it automatically adds a default `mule-app.properties` file that you can use to add your properties for the application. However, in Mule 4 this file is no longer automatically created. +

Reviewer: ```suggestion
In Mule 4, creating a new application via Anypoint Studio no longer automatically adds the default `mule-app.properties` file added in Mule 3. +
```

If you choose to create a custom `.properties` file, you also need to configure it in your Mule app via the Configuration Properties Global Element. The XML configuration looks like this:

[source,xml,linenums]
----
<configuration-properties file="myConfiguration.properties"/>
----

> MuleSoft highly recommends that you do not package the configuration files for all the environments inside your app. Instead, you should use a `.yaml` file to provide defaults, then use Runtime Manager to override each of these properties at deployment time.

Reviewer: ```suggestion
Best practices dictate that you not package the configuration files for all the environments inside your app. Instead, use a `.yaml` file to provide defaults, and then use Runtime Manager to override each of these properties at deployment.
```
